eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -0,0 +1,350 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Created by jmmut on 2015-10-28.
+ *
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+public class VariantExporter {
+
+    private static final Logger logger = LoggerFactory.getLogger(VariantExporter.class);
+    private int failedVariants = 0;
+
+    /**
+     *
+     * @param iterator where to get the variants from
+     * @param outputDir directory to write the output vcf(s).
+     * @param sourceDBAdaptor to retrieve all the VariantSources in any VariantSourceEntry.
+     * @param options not implemented yet, use only for studyId and fileId
+     * @return num variants not written due to errors
+     */
+    public List<String> VcfHtsExport(Iterator<Variant> iterator, String outputDir,
+                                            VariantSourceDBAdaptor sourceDBAdaptor, QueryOptions options) throws IOException {
+
+
+        List<String> studyIds = options.getAsStringList(VariantDBAdaptor.STUDIES);
+        List<VariantSource> sourcesList = sourceDBAdaptor.getAllSourcesByStudyIds(studyIds, options).getResult();
+
+        Map<String, VariantSource> sources = new TreeMap<>();
+
+        for (VariantSource variantSource : sourcesList) {
+            sources.put(variantSource.getStudyId(), variantSource);
+        }
+        
+        Map<String, VCFHeader> headers = getVcfHeaders(sources);
+//        header.addMetaDataLine(new VCFFilterHeaderLine("PASS", "Valid variant"));
+//        header.addMetaDataLine(new VCFFilterHeaderLine(".", "No FILTER info"));
+
+        String suffix = ".exported.vcf.gz";
+        List<String> files = new ArrayList<>();
+        Map<String, VariantContextWriter> writers = new TreeMap<>();
+
+        // setup writers
+        for (String studyId : studyIds) {
+            if (!headers.containsKey(studyId)) {
+                throw new IllegalArgumentException("aborting VCF export: missing header for study " + studyId);
+            }
+            VariantContextWriterBuilder builder = new VariantContextWriterBuilder();
+            File outFile = Paths.get(outputDir).resolve(studyId + suffix).toFile();
+            files.add(outFile.getPath());
+            VariantContextWriter writer = builder
+                    .setOutputFile(outFile)
+                    .setReferenceDictionary(headers.get(studyId).getSequenceDictionary())
+                    .unsetOption(Options.INDEX_ON_THE_FLY)
+                    .build();
+            writers.put(studyId, writer);
+            writer.writeHeader(headers.get(studyId));
+        }
+
+        // actual loop
+        int failedVariants = 0;
+        StringBuilder filesConcat = new StringBuilder();
+        for (String file : files) {
+            filesConcat.append(file).append(" ");
+        }
+        logger.info("Exporting to files: [ " + filesConcat.toString() + "]");
```

Comment:
This loop could be replaced by ```StringUtils.join(files, ", ")``` in the log message (biodata-formats includes apache-commons-lang anyway).